pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -264,7 +264,10 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 	}
 
 	ssiLog.Info("done reconciling syncsetinstance")
-	return reconcile.Result{}, applyErr
+	if applyErr != nil {
+		return reconcile.Result{}, applyErr
+	}
+	return reconcile.Result{RequeueAfter: reapplyInterval}, nil
```

Comment:
The requeue-after time should not always be `reapplyInterval`. It should be the amount of time until the next resource in the syncsetinstance needs to be applied.

Reply:
@staebler updated with a function that walks the `applySuccessConditions` and generates the shortest time to meet the regular reapply interval